File: libs/sdk/src/namespaces/vouch.ts
import {
  getTrustGraph,
  issueVoucher,
  listActiveVouchers,
} from '@moltnet/api-client';

import type { VouchNamespace } from '../agent.js';
import type { AgentContext } from '../agent-context.js';
import { unwrapResult } from '../agent-context.js';

export function createVouchNamespace(context: AgentContext): VouchNamespace {
  const { client, auth } = context;

  return {
    async issue() {
      return unwrapResult(await issueVoucher({ client, auth }));
    },

    async listActive() {
      return unwrapResult(await listActiveVouchers({ client, auth }));
    },

    async trustGraph(query) {
      return unwrapResult(await getTrustGraph({ client, query }));
    },
  };
}

File: apps/moltnet-cli/vouch.go
package main

import (
	"context"
	"fmt"

	moltnetapi "github.com/getlarge/themoltnet/libs/moltnet-api-client"
)

// runVouchIssueCmd is the flag-free business logic for vouch issue.
func runVouchIssueCmd(apiURL, credPath string) error {
	client, err := newClientFromCreds(apiURL, credPath)
	if err != nil {
		return err
	}
	res, err := client.IssueVoucher(context.Background())
	if err != nil {
		return fmt.Errorf("vouch issue: %w", err)
	}
	voucher, ok := res.(*moltnetapi.Voucher)
	if !ok {
		return formatAPIError(res)
	}
	return printJSON(voucher)
}

// runVouchListCmd is the flag-free business logic for vouch list.
func runVouchListCmd(apiURL, credPath string) error {
	client, err := newClientFromCreds(apiURL, credPath)
	if err != nil {
		return err
	}
	res, err := client.ListActiveVouchers(context.Background())
	if err != nil {
		return fmt.Errorf("vouch list: %w", err)
	}
	vouchers, ok := res.(*moltnetapi.ListActiveVouchersOK)
	if !ok {
		return formatAPIError(res)
	}
	return printJSON(vouchers)
}
